### DISCUSSION TITLE
Use accessor in BelongsToManyMultiSelect field

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#1384] [Help] [answered]
It would be great if you could use accessors in relationship method for BelongsToManyMultiSelect field.

For example, I would like to select users for some resources (which is quite common need) by their "fullname", but in database this information is stored in 2 separate columns: "firstname" and "lastname". Using only one of this columns is not accurate enough.

```php
BelongsToManyMultiSelect::make('users')
	->relationship('users', 'fullname'),
```

**Comment by danharrin** [comment, marked as answer]
Hey, I suggest using a virtual column in MySQL to concatenate the first and last names.

https://mydbops.wordpress.com/2017/07/19/virtual-columns-in-mysql-and-use-cases

The reason why it doesn't work out of the box is because we use Eloquent queries on the backend, which don't work with accessors. You could always use a custom `MultiSelect` component though to work around this, but the virtual column is easier.

**Reply by studiowizjo** [reply]
Thank you. Works great. Here is a complete solution for future reference:

**SQL query:**
```sql
ALTER TABLE users ADD full_name varchar(100) GENERATED ALWAYS AS (concat(firstname,' ',lastname)) VIRTUAL NOT NULL AFTER lastname
```

Besides that it is necessary to remove accessor (in this example: "getFullNameAttribute") from model.

**Reply by fauzie811** [reply]
FYI, you can also use migration to create a virtual column:

```php
$table->string('full_name')->virtualAs("CONCAT(firstname,' ',lastname)")->after('lastname');
```

**Reply by studiowizjo** [reply]
Even better. Thank you :).
